### PR TITLE
Add some route-based VPN test scenarios

### PR DESCRIPTION
**Pull request by Xenopathic**
To complement https://wiki.strongswan.org/projects/strongswan/wiki/RouteBasedVPN, I've created some route-based VPN test scenarios. I also fixed a small issue with the PATH environment variable in the testing build chroot, which was stopping me from running the tests on my machine.

Let me know if putting these tests in a new directory was the wrong decision, and they should instead be distributed amongst {ikev2,ipv6}. I may add more tests later, e.g. VTI6 tests (IPv6 IPsec), or GRE tests.

### CODE REVIEW

File: testing/config/kernel/config-3.10 [hunk 1]
```diff
@@ -477,7 +477,7 @@ CONFIG_IP_ROUTE_CLASSID=y
 CONFIG_NET_IP_TUNNEL=y
 # CONFIG_ARPD is not set
 # CONFIG_SYN_COOKIES is not set
-# CONFIG_NET_IPVTI is not set
```
[the remark this answers is not in the document]
Comment (Xenopathic) [reply testing/config/kernel/config-3.10]: I updated testing.conf to point at the latest 4.13 kernel, and only added CONFIG_NET_IPVTI for that config. I also took the config from the build dir (after oldconfig) but it doesn't seem to have set CONFIG_IPV6_VTI, so perhaps they aren't connected?

**Comment by Xenopathic** [comment]
Along with adding a net2net-vti scenario, I changed the other scenarios to use PH_IP_* instead of hardcoded IP addresses. I know there's a lot of technical debt regarding that anyway, but I figured it's easier to get it right for these scenarios now than leave it

File: testing/config/kernel/config-3.10 (hunk 1 above)
[the remark this answers is not in the document]
Comment (tobiasbrunner) [reply testing/config/kernel/config-3.10]: Yeah that's possible, I just now saw that your ip6 scenario is actually an ip6-in-ip4 scenario, i.e. IPv6 VTI devices are not required due to the IPv4 tunnel addresses. Would probably be a good idea to clarify this in the name (as done in the ipv6 scenarios).

**Comment by Thermi** [comment]
GRE tunnels are very simple.
You create one via "ip tunnel [...]" using a key that is the same on the other side. Then you use IPsec to protect it (the GRE protocol) between the two hosts. You obviously do not need to protect only GRE. Just route your networks over the tunnel device (just a simple network interface). It is a layer 3 tunnel though, so you don't need to set any next hops.

**Comment by tobiasbrunner** [comment]
Great, thanks! I've pushed the commits, with some small changes (e.g. renaming the scenario to _...ip6-in-ip4_, fixing some checks in evaltest.dat - which were actually wrong in several other swanctl-based scenarios, disabling the policy lookup on the VTI device), to the _testing-route-based_ branch. I also added a site-to-site GRE scenario (basically a combination of your swanctl-based _net2net-vti_ scenario and my original _net2net-gre_ scenario).

**Comment by tobiasbrunner** [comment]
Merged to master. Thanks again!

**Pull request closed without merging**